static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx

```diff
@@ -19,23 +19,36 @@ import {t, tct} from 'sentry/locale';
 import {SentryFunction} from 'sentry/types';
 import useApi from 'sentry/utils/useApi';
 
+import SentryFunctionEnvironmentVariables from './sentryFunctionsEnvironmentVariables';
 import SentryFunctionSubscriptions from './sentryFunctionSubscriptions';
 
 class SentryFunctionFormModel extends FormModel {
+  getEnvVariables: null | (() => EnvVariable[]) = null;
+  getEventHooks: null | (() => string[]) = null;
+
   getTransformedData() {
     const data = super.getTransformedData() as Record<string, any>;
-    const events: string[] = [];
-    if (data.onIssue) {
-      events.push('issue');
-    }
-    if (data.onError) {
-      events.push('error');
-    }
-    if (data.onComment) {
-      events.push('comment');
-    }
-    data.events = events;
+    // const events: string[] = [];
+    // if (data.onIssue) {
+    //   events.push('issue');
+    // }
+    // if (data.onError) {
+    //   events.push('error');
+    // }
+    // if (data.onComment) {
+    //   events.push('comment');
+    // }
+    // data.events = events;
     const {...output} = data;
+    for (const key in data) {
+      if (key.startsWith('env-variable')) {
```

Comment:
@vpandiarajan20 wait we have the env variables in the data? Why don’t we use that instead?